hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AuthenticationBaseMessageTask.java

```diff
@@ -134,6 +134,12 @@ public void processMessage() throws Throwable {
             return;
         }
         String uuid = getUuid();
+        if (clientMessage.getCorrelationId() < clientEngine.getLastAuthenticationCorrelationId(uuid)) {
```

Comment:
Isn't this already checked by `ClientReAuthOperation`? Is this a requirement or optimization? This one duplicates the logic in `clientEngine`.